ReactSkia/sdk/CurlNetworking.cpp

```diff
@@ -88,6 +106,23 @@ void CurlNetworking::processNetworkRequest(CURLM *curlMultiHandle) {
             (curlRequest->curlResponse.responseTimeout = true)
             :(curlRequest->curlResponse.responseTimeout = false);
         curl_multi_remove_handle(curlMultiHandle, curlHandle);
+        char *url = NULL;
+        curl_easy_getinfo(curlHandle, CURLINFO_EFFECTIVE_URL, &url);
+        canCacheData_ = true;
+        cacheExpiryTime_ = DEFAULT_MAX_CACHE_EXPIRY_TIME;
+        double responseMaxAgeTime = DEFAULT_MAX_CACHE_EXPIRY_TIME;
+        double requestMaxAgeTime = DEFAULT_MAX_CACHE_EXPIRY_TIME;
+        // Parse server response headers and retrieve caching details
+        auto responseCacheControlData = curlRequest->curlResponse.headerBuffer.find("Cache-Control");
+        if(responseCacheControlData != curlRequest->curlResponse.headerBuffer.items().end()) {
+          std::string responseCacheControlString = responseCacheControlData->second.asString();
+          canCacheData_ = shouldCacheData(responseCacheControlString);
+          if(canCacheData_){
+           responseMaxAgeTime = getCacheMaxAgeDuration(responseCacheControlString);
```

Comment:
This function can be combined with shouldCacheData and pass max age as out parameter.